File: php/Film3D.php
<?php
class Film3D extends Film{
    // atribut
    private $formatLayar;
    private $biayaKacamata;
    private $jumlahKacamataTersedia;

    // konstruktor
    public function __construct($id, $judul, $durasi, $jamTayang, $hargaTiket,
                    $genre, $sutradara, $ratingUsia, $bahasa, $tahunRilis,
                    $formatLayar, $biayaKacamata, $jumlahKacamataTersedia){
        parent:: __construct($id, $judul, $durasi, $jamTayang, $hargaTiket,
                    $genre, $sutradara, $ratingUsia, $bahasa, $tahunRilis);
        $this->formatLayar = $formatLayar;
        $this->biayaKacamata = $biayaKacamata;
        $this->jumlahKacamataTersedia = $jumlahKacamataTersedia;
    }

    // getter
    public function getFormatLayar(){
        return $this->formatLayar;
    }
    public function getBiayaKacamata(){
        return $this->biayaKacamata;
    }
    public function getJumlahKacamataTersedia(){
        return $this->jumlahKacamataTersedia;
    }

    // setter
    public function setFormatLayar($formatLayar){
        $this->formatLayar = $formatLayar;
    }
    public function setBiayaKacamata($biayaKacamata){
        $this->biayaKacamata = $biayaKacamata;
    }
    public function setJumlahKacamataTersedia($jumlahKacamataTersedia){
        $this->jumlahKacamataTersedia = $jumlahKacamataTersedia;
    }
}


?>

File: python/Film3D.py
from Film import Film

class Film3D(Film):
    def __init__(self, id, judul, durasi, jamTayang, hargaTiket, 
                 genre, sutradara, ratingUsia, bahasa, tahunRilis, 
                 formatLayar, biayaKacamata, jumlahKacamataTersedia):
        super().__init__(id, judul, durasi, jamTayang, hargaTiket, 
                         genre, sutradara, ratingUsia, bahasa, tahunRilis)
        self.formatLayar = formatLayar
        self.biayaKacamata = biayaKacamata
        self.jumlahKacamataTersedia = jumlahKacamataTersedia

    # getter
    def get_formatLayar(self):
        return self.formatLayar
    def get_biayaKacamata(self):
        return self.biayaKacamata
    def get_jumlahKacamataTersedia(self):
        return self.jumlahKacamataTersedia

    # setter
    def set_formatLayar(self, formatLayar):
        self.formatLayar = formatLayar
    def set_biayaKacamata(self, biayaKacamata):
        self.biayaKacamata = biayaKacamata
    def set_jumlahKacamataTersedia(self, jumlahKacamataTersedia):
        self.jumlahKacamataTersedia = jumlahKacamataTersedia